File: c-sharp/recursion/ReverseLinkedList.cs
namespace c_sharp.recursion
{
    public class ReverseLinkedList
    {
        public static ListNode Reverse(ListNode head)
        {
            // create a new node to store null
            ListNode prev = null;

            while (head != null)
            {
                // store next of head in a temp var
                var temp = head.next;

                // set next of head tp previous
                head.next = prev;

                // move your prev to current head
                prev = head;

                // set head to temp for next iteration
                head = temp;
            }

            return prev;
        }
    }
}

File: java/src/com/rohitsh/recursion/ReverseLinkedList.java
package com.rohitsh.recursion;

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

public class ReverseLinkedList {

    public ListNode reverseList(ListNode head) {

        ListNode prev = null;

        while(head != null){

            // store next node
            var temp = head.next;

            // set current head to previous
            head.next = prev;

            // set head to temp
            prev = head;

            // set head to next node
            head = temp;
        }

        return prev;
    }
}
